frontend/packages/dev-console/src/components/buildconfig/BuildConfigFormEditor.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import { useFormikContext } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import SwitchToYAMLAlert from '@console/shared/src/components/alerts/SwitchToYAMLAlert';
+import FormSection from '../import/section/FormSection';
+import ProgressiveList from '../progressive-list/ProgressiveList';
+import ProgressiveListItem from '../progressive-list/ProgressiveListItem';
+import EnvironmentVariablesSection from './sections/EnvironmentVariablesSection';
+import HooksSection from './sections/HooksSection';
+import ImagesSection from './sections/ImagesSection';
+import NameSection from './sections/NameSection';
+import PolicySection from './sections/PolicySection';
+import SecretsSection from './sections/SecretsSection';
+import SourceSection from './sections/SourceSection';
+import TriggersSection from './sections/TriggersSection';
+
+export type BuildConfigFormEditorProps = {
+  resourceObj: K8sResourceKind;
+};
+
+const BuildConfigFormEditor: React.FC<BuildConfigFormEditorProps> = ({ resourceObj }) => {
+  const { t } = useTranslation();
+  const [showYAMLAlert, setShowYAMLAlert] = React.useState<boolean>(true);
+
+  const formik = useFormikContext();
+
+  const [visibleItems, setVisibleItems] = React.useState<string[]>([]);
+  const handleVisibleItemChange = (item: string) => {
+    setVisibleItems([...visibleItems, item]);
+  };
+
+  const namespace = resourceObj?.metadata?.namespace;
+
+  return (
+    <>
+      {showYAMLAlert && <SwitchToYAMLAlert onClose={() => setShowYAMLAlert(false)} />}
+
+      <NameSection />
+      <SourceSection />
+      <ImagesSection />
+      <EnvironmentVariablesSection namespace={namespace} />
+
+      <FormSection
+        title={t('devconsole~Advanced options')}
+        dataTest="section advanced-options"
+        fullWidth
+      >
+        <ProgressiveList
+          text={t('devconsole~Click on the names to access advanced options for')}
+          visibleItems={visibleItems}
+          onVisibleItemChange={handleVisibleItemChange}
+        >
+          <ProgressiveListItem name={t('devconsole~Triggers')}>
+            <TriggersSection namespace={namespace} />
+          </ProgressiveListItem>
+          <ProgressiveListItem name={t('devconsole~Secrets')}>
+            <SecretsSection namespace={namespace} />
+          </ProgressiveListItem>
+          <ProgressiveListItem name={t('devconsole~Run Policy')}>
+            <PolicySection />
+          </ProgressiveListItem>
+          <ProgressiveListItem name={t('devconsole~Hooks')}>
+            <HooksSection />
+          </ProgressiveListItem>
+        </ProgressiveList>
+      </FormSection>
```

Comment:
Create a separate component for `Advanced options`

Reply:
Thanks for that idea, but I like that all "sections" are listed in one file and that this gives a code-reader an instant overview which sections, which are part of the advanced options and also which order to we use for the directly visible sections and the advanced options. I would like to keep this as it is.